swo/mpt/cli/core/accounts/app.py

```diff
@@ -0,0 +1,225 @@
+import typer
+from rich.table import Table
+from swo.mpt.cli.core.console import console, status_progress
+from swo.mpt.cli.core.errors import AccountNotFoundError, MPTAPIError
+from swo.mpt.cli.core.mpt.client import MPTClient
+from swo.mpt.cli.core.mpt.flows import get_token
+from typing_extensions import Annotated
+
+from .flows import (
+    disable_accounts_except,
+    find_account,
+    from_token,
+    get_accounts_file_path,
+    get_or_create_accounts,
+    is_account_exists,
+    remove_account,
+    write_accounts,
+)
+from .models import Account
+
+app = typer.Typer()
+
+
+@app.command(name="add")
+def add_account(
+    token_id: Annotated[
+        str,
+        typer.Argument(help="SoftwareOne Marketplace API Token ID", metavar="TOKEN-ID"),
+    ],
+    secret: Annotated[
+        str, typer.Argument(help="SoftwareOne Marketplace API Token secret")
+    ],
+    environment: Annotated[
+        str, typer.Option("--environment", "-e", help="URL to the API for environment")
+    ] = "https://api.platform.softwareone.com/v1",
+):
+    """
+    Add account to work with the SoftwareOne Marketplace
+    """
+    with status_progress() as (status, _):
```

Comment:
Please take a look also to https://rich.readthedocs.io/en/latest/_modules/rich/status.html#Status and to https://github.com/softwareone-platform/swo-adobe-vipm-extension/blob/main/devmock/mockgendata.py for an alternative way of rendering an undetermined progress.